src/stochprog.jl

```diff
@@ -0,0 +1,112 @@
+using LightGraphs
+
+export AbstractStochasticProgram, stochasticprogram
+export isleaf, probability, cutgenerator, numberofpaths
+export setprobability!, add_scenario_state!, add_scenario_transition!
+
+"""
+    AbstractStochasticProgram
+
+Stochastic program instance
+"""
+abstract type AbstractStochasticProgram <: AbstractGraph{Int} end
+
+"""
+    stochasticprogram(args...)
+
+Creates a stochastic program from the arguments
+"""
+function stochasticprogram end
+
+"""
+    getmaster(sp::AbstractStochasticProgram)
+
+Returns the master node of stochastic problem `sp`.
+"""
+function getmaster end
+
+"""
+    probability(sp::AbstractStochasticProgram, edge)
+
+Returns the probability to take the edge `edge` in the stochastic problem `sp`.
+"""
+function probability end
+
+"""
+    setchildx!(sp::AbstractStochasticProgram, node, child, sol)
+
+Sets the parent solution of `child` as `sol`, the solution obtained at `node`.
+"""
+function setchildx! end
+
+"""
+    solve!(sp::AbstractStochasticProgram, node)
+
+Solves the program at node `node` in `sp`.
+"""
+function solve! end
+
+"""
+    getobjlb(sp::AbstractStochasticProgram, node)
+
+Gets the current lower bound to the objective of `node`.
+"""
+function getobjlb end
```

Comment:
I think we should aim for readability over conciseness `getobjectivebound`? Because it might be maximization

Reply:
I agree, that's better :)

Reply:
Done iin 9dafe36